components/eventing-controller/pkg/backend/jetstreamv2/jetstream_internal_unit_test.go

```diff
@@ -0,0 +1,425 @@
+func Test_CheckNATSSubscriptionsCount(t *testing.T) {
+	// given
+	subWithType := subtesting.NewSubscriptionWithOneType()
+	defaultLogger, err := logger.New(string(kymalogger.JSON), string(kymalogger.INFO))
+	require.NoError(t, err)
+	jsCleaner := cleaner.NewJetStreamCleaner(defaultLogger)
+	jsBackend := &JetStream{
+		subscriptions: make(map[SubscriptionSubjectIdentifier]Subscriber),
+		cleaner:       jsCleaner,
+	}
+
+	testCases := []struct {
+		name                 string
+		givenSubscription    *v1alpha2.Subscription
+		givenSubscriptionMap map[SubscriptionSubjectIdentifier]Subscriber
+		wantErr              error
+	}{
+		{
+			name:                 "empty subscriptions map with subscription with no types should result in not error",
+			givenSubscription:    subtesting.NewSubscriptionWithEmptyTypes(),
+			givenSubscriptionMap: map[SubscriptionSubjectIdentifier]Subscriber{},
+			wantErr:              nil,
+		},
+		{
+			name:              "if the subscriptions map contains all the NATS Subscriptions, no error is expected",
+			givenSubscription: subWithType,
+			givenSubscriptionMap: map[SubscriptionSubjectIdentifier]Subscriber{
+				NewSubscriptionSubjectIdentifier(subWithType, "kyma./default/kyma/id.prefix.testapp1023.order.created.v1"): &nats.Subscription{},
+			},
+			wantErr: nil,
+		},
+		{
+			name:                 "unexpected empty subscriptions map should result into an error",
+			givenSubscription:    subWithType,
+			givenSubscriptionMap: map[SubscriptionSubjectIdentifier]Subscriber{},
+			wantErr:              &backenderrors.ErrMissingNATSSubscription{},
+		},
+	}
+
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+
+			// inject the fake subscription map
+			jsBackend.subscriptions = testCase.givenSubscriptionMap
+
+			// when
+			err := jsBackend.checkNATSSubscriptionsCount(testCase.givenSubscription)
+
+			// then
+			if testCase.wantErr == nil {
+				require.NoError(t, err)
+			} else {
+				require.ErrorIs(t, err, testCase.wantErr)
+			}
+		})
+	}
+
+}
+
+// HELPER FUNCTION
+
+// sets up the TestEnvironment with the mocks instead of running a full-fledged NATS test server.
+func getTestEnvironmentWithMock(t *testing.T) *TestEnvironment {
+	defaultLogger, err := logger.New(string(kymalogger.JSON), string(kymalogger.INFO))
+	require.NoError(t, err)
+	jsCtx := &mocks.JetStreamContext{}
+	natsConfig := defaultNatsConfig("localhost")
+	metricsCollector := metrics.NewCollector()
+
+	jsCleaner := cleaner.NewJetStreamCleaner(defaultLogger)
+	jsBackend := &JetStream{
+		Config:           natsConfig,
+		Conn:             nil,
+		jsCtx:            jsCtx,
+		subscriptions:    make(map[SubscriptionSubjectIdentifier]Subscriber),
+		sinks:            sync.Map{},
+		logger:           defaultLogger,
+		metricsCollector: metricsCollector,
+		cleaner:          jsCleaner,
+	}
+
+	return &TestEnvironment{
+		jsBackend:  jsBackend,
```

Comment:
Isn't is enough to just return the `JetStream` object since this is the `System under Test` ?